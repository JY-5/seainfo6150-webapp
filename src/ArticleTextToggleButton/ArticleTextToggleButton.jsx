import React from "react";
import styles from "./ArticleTextToggleButton.module.css";
import PropTypes from "prop-types";

const ArticleTextToggleButton = (props) => {
    return (
        <button onClick={props.onClick} className={styles.toggleButton}>{props.buttonText}</button>
    )
};

ArticleTextToggleButton.propTypes = {
    buttonText: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired,
};
export default ArticleTextToggleButton;