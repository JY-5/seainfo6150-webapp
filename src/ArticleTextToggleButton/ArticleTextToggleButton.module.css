button {
    color: white;
    background-color: #c30202;
    margin-bottom: 1rem;
    padding: 0.5rem;
    box-shadow: inset 5px 5px 10px 10px #c30202;
    border-radius: 10px;
}