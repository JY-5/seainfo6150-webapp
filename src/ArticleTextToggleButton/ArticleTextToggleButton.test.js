import React from "react";
import ArticleTextToggleButton from "./ArticleTextToggleButton";

describe("ArticleTextToggleButton tests", () => {
    it("renders correctly", () => {
        const { container, getByText } = render(<ArticleTextToggleButton buttonText="Show more" onClick={() => {}} />);
        expect(container).toMatchSnapshot();
    });
});