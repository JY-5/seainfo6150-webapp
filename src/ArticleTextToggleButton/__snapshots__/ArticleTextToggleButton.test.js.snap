// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleTextToggleButton tests renders correctly 1`] = `
<div>
  <button
    class="toggleButton"
  >
    Show more
  </button>
</div>
`;
